### ISSUE TITLE
RoPE doesn't work

### ISSUE THREAD

**Issue by imhuim982** [ByteDance-Seed/VeOmni#414]
### System Info

When my model is initializing, the inv_freq seems right. But during training, the inv_freq all become zero!

<img width="1684" height="780" alt="Image" src="https://github.com/user-attachments/assets/b94b87a7-943a-4b35-b185-303236d8eac1" />

<img width="2076" height="776" alt="Image" src="https://github.com/user-attachments/assets/0c718e24-3ca1-4210-9b6d-8207e9e46b4e" />

### Information

- [ ] The official example scripts
- [x] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [x] My own task or dataset (give details below)

### Reproduction

1. Train a qwen3 model using the standard pretrain script.
2. print rope inv_freq

### Expected behavior

1. inv_freq not all zero during training

**Comment by imhuim982** [comment]
I'm sure something wrong with build_parallelize_model

<img width="2027" height="795" alt="Image" src="https://github.com/user-attachments/assets/fd94aa1d-5162-4da6-af0b-eb6e90b54910" />

**Comment by imhuim982** [comment]
Here is my training script:

```
torchrun --nnodes=1 --nproc-per-node=2 --node-rank=0 tasks/pretrain.py configs/baselines/full_attn_tiny_cos.yaml \
    --model.config_path configs/full_attn_tiny/config.json \
    --data.train_path /my_data \
    --data.max_seq_len 8192 \
    --data.train_size 10000000000 \
    --train.init_device meta \
    --train.use_wandb true \
    --train.enable_gradient_checkpointing true \
    --train.rmpad false \
    --train.rmpad_with_pos_ids false \
    --train.enable_mixed_precision \
    --train.micro_batch_size 1 \
    --train.global_batch_size 2 \
    --train.lr 3e-4 \
    --train.lr_min 3e-5 \
    --train.ulysses_parallel_size 1 \
    --train.save_steps 10000 \
    --train.max_steps 30000 \
    --train.load_checkpoint_path auto \
    --train.output_dir ../checkpoints/rope-debug
```

full_attn_tiny_cos.yaml looks like:
```
model:
  tokenizer_path: configs/olmo3_vocab
  attn_implementation: flash_attention_3

data:
  train_size: 1000000000000
  dataloader_type: native
  data_type: plaintext
  max_seq_len: 4096
  text_keys: text
  drop_last: true

train:
  data_parallel_mode: fsdp2
  ulysses_parallel_size: 1
  global_batch_size: 16
  micro_batch_size: 2
  rmpad: false
  rmpad_with_pos_ids: true
  bsz_warmup_ratio: 0.007
  dyn_bsz_margin: 0
  dyn_bsz_buffer_size: 200
  optimizer: adamw
  lr: 2.0e-3
  lr_warmup_ratio: 0.01
  lr_decay_style: cosine
  weight_decay: 0.1
  max_grad_norm: 1.0
  enable_mixed_precision: true
  enable_gradient_checkpointing: false
  enable_full_shard: true
  enable_fsdp_offload: false
  enable_activation_offload: false
  init_device: meta
  enable_full_determinism: false
  empty_cache_steps: 500
  ckpt_manager: dcp
  save_steps: 100
  save_hf_weights: true
  wandb_project: FullAttnBaseline
```

my model config: 
```
{
    "architectures": [
      "Qwen2ForCausalLM"
    ],
    "attention_dropout": 0.0,
    "bos_token_id": 100257,
    "eos_token_id": 100257,
    "hidden_act": "silu",
    "hidden_size": 1024,
    "initializer_range": 0.02,
    "intermediate_size": 4096,
    "max_position_embeddings": 131072,
    "max_window_layers": 24,
    "model_type": "qwen3",
    "num_attention_heads": 16,
    "num_hidden_layers": 24,
    "num_key_value_heads": 4,
    "rms_norm_eps": 1e-06,
    "rope_theta": 1000000.0,
    "sliding_window": 131072,
    "tie_word_embeddings": true,
    "torch_dtype": "bfloat16",
    "transformers_version": "4.40.1",
    "use_cache": true,
    "use_sliding_window": false,
    "vocab_size": 100278,
    "_attn_implementation": "flash_attention_3"
  }
```

**Comment by Coach257** [comment]
I think it might because that this registered buffer is not saved in ckpt, as it is always initialized from init_rope_fn. When using init_device=meta, it fail to load rope from ckpt and become all zeros.

**Comment by imhuim982** [comment]
> I think it might because that this registered buffer is not saved in ckpt, as it is always initialized from init_rope_fn. When using init_device=meta, it fail to load rope from ckpt and become all zeros.

The inv_freq looks fine when printed before the build_parallelize_model call. It doesn't seem to be a loading problem; the RoPE values are all being initialized from the config anyway.

**Comment by Luosuu** [comment]
thanks for reporting @imhuim982 
I will investigate this issue soon.

**Comment by Luosuu** [comment]
Hi @imhuim982 
I could not reproduce your issue..

<img width="1627" height="203" alt="Image" src="https://github.com/user-attachments/assets/05fb3846-a91a-46da-95c9-8648336f5182" />

**Comment by imhuim982** [comment]
Hi @Luosuu 

Could you kindly clarify which version you are working on?  I'm working on v1.5.0. I noticed that the implementation of modeling_qwen3 in the current main branch seems different from previous ones. The version I'm using has the complete implementation of modeling_qwen3.

**Comment by Luosuu** [comment]
@imhuim982 Hi,  I am on the latest commit.

**Comment by imhuim982** [comment]
Thanks for the update. I'm unable to verify the issue on the main branch at the moment due to my current setup. I'm using a temporary patch for now, but I will close this issue as you mentioned the main branch works fine.
